featuretools/primitives/standard/transform_primitive.py

```diff
@@ -584,49 +587,75 @@ def email_address_to_domain(emails):
         return email_address_to_domain
 
 
-class NumericLag(TransformPrimitive):
+class Lag(TransformPrimitive):
```

Comment:
What if a user is already be using the `NumericLag` primitive?

Reply:
Good question, I actually wanted to bring this up. Do we have users using featuretools for time series currently? If not I say just go ahead and replace it but if not we can just revert this and add a DeprecationWarning and remove it for V2. Thoughts?

Reply:
We could add a DeprecationWarning  if the user uses `NumericLag`, but still get `Lag` primitive into Featuretools. 
OR, we could update `dfs` to support the old primitive name. 

@rwedge any suggestion here?

Reply:
I'm in favor of deprecating `NumericLag`.  I think updating `dfs` to map "numeric_lag" to the `Lag` primitive would be problematic.  `Lag` has more accepted inputs and could create more features than the user was expecting, if we magically swapped in `Lag` for them.  It also would hinder custom primitives from using "numeric_lag" for as long as we have this `dfs` shortcut, even after we remove the original primitive

Reply:
Added a `FutureWarning` to the primitive. Similar to how pandas does their warnings.

Reply:
Can we add a test checking the warning works as expected?

Reply:
Done